### DISCUSSION TITLE
Application access: "Forbidden"

### DISCUSSION THREAD

**Discussion by KasperSkytte** [gravitational/teleport#13164] [Q&A] [answered]
I'm having an issue with serving some applications through teleport. When I launch the apps I only see "Forbidden". In the logs I see messages like:
`User Message: this request can be only executed by a node, proxy or kube service] filesessions/fileasync.go:228`
`WARN [APP:SERVI] Failed to serve request: this request can be only executed by a node, proxy or kube service. app/server.go:580`

From one of my nodes I do serve an app succesfully. The apps used to work fine, but stopped working at some point. I may have either upgraded teleport, recreated my user account to add another login, or rotated CA . I didn't notice before a while after. I have service files set up to manage the apps using a separate config file for each app. I tried enabling proxy service in the configs, updating ca_pin, new auth token and some more. I'm not sure where to adjust things or whether I should reinstall and configure teleport everywhere. I'm the only user currently.

Why can't I access apps?

**Comment by webvictim** [comment, marked as answer]
I think was a bug that existed in a few earlier v9 versions of Teleport. If you update to 9.3+ it should work as expected.

**Reply by KasperSkytte** [reply]
Thanks. Upgrading server from 9.1.0 to 9.3.4 fixed the issue.

**Reply by webvictim** [reply]
No worries. Sorry for the hassle.
